### ISSUE TITLE
Error when starting with micromatch package

### ISSUE THREAD

**Issue by jonas-sk** [netlify-templates/one-click-hugo-cms#125]
The following error appears when I `npm start`:

    C:\Users\Jonas\Desktop\test\one-click-hugo-cms [master ≡]> npm start    

    > victor-hugo@1.0.0 start C:\Users\Jonas\Desktop\test\one-click-hugo-cms
    > gulp server   

    [17:00:43] Failed to load external module @babel/register
    [17:00:43] Requiring external module babel-register
    C:\Users\Jonas\Desktop\test\one-click-hugo-cms\node_modules\micromatch\index.js:44
        let isMatch = picomatch(String(patterns[i]), { ...options, onResult }, true);
                                                       ^^^
    SyntaxError: Unexpected token ...
        at createScript (vm.js:53:10)
        at Object.runInThisContext (vm.js:95:10)
        at Module._compile (module.js:543:28)
        at Module._extensions..js (module.js:580:10)
        at Object.require.extensions.(anonymous function) [as .js] (C:\Users\Jonas\Desktop\test\one-click-hugo-cms\node_modules\babel-register\lib\node.js:152:7)
        at Module.load (module.js:488:32)
        at tryModuleLoad (module.js:447:12)
        at Function.Module._load (module.js:439:3)
        at Module.require (module.js:498:17)
        at require (internal/module.js:20:19)   

    npm ERR! Windows_NT 10.0.17763
    npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
    npm ERR! node v7.7.2
    npm ERR! npm  v4.1.2
    npm ERR! code ELIFECYCLE
    npm ERR! victor-hugo@1.0.0 start: `gulp server`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the victor-hugo@1.0.0 start script 'gulp server'.
    npm ERR! Make sure you have the latest version of node.js and npm installed.
    npm ERR! If you do, this is most likely a problem with the victor-hugo package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     gulp server
    npm ERR! You can get information on how to open an issue for this project with:
    npm ERR!     npm bugs victor-hugo
    npm ERR! Or if that isn't available, you can get their info via:
    npm ERR!     npm owner ls victor-hugo
    npm ERR! There is likely additional logging output above.   

    npm ERR! Please include the following file with any support request:
    npm ERR!     C:\Users\Jonas\Desktop\test\one-click-hugo-cms\npm-debug.log

Steps to reproduce
1. `git clone` the repo
2. `npm install`
3. `npm start`
4. Get the error

The `micromatch` version throwing the error is 4.0.2. What packages require it:

    +-- browser-sync@2.26.4
    | +-- chokidar@2.1.5
    | | +-- anymatch@2.0.0
    | | | `-- micromatch@3.1.10
    | | `-- readdirp@2.2.1
    | |   `-- micromatch@3.1.10
    | `-- micromatch@4.0.2
    +-- gulp@3.9.1
    | `-- liftoff@2.5.0
    |   `-- findup-sync@2.0.0
    |     `-- micromatch@3.1.10
    +-- svg-sprite-loader@3.9.2
    | `-- svg-baker@1.4.0
    |   `-- micromatch@3.1.0
    `-- UNMET PEER DEPENDENCY webpack@3.12.0    

    npm ERR! peer dep missing: webpack@^4.0.0, required by css-loader@2.1.1

**Comment by erquhart** [comment]
Not able to reproduce :/

**Comment by jonas-sk** [comment]
Yes. I just retried. This is the output of a fresh `git clone`:

    C:\Users\Jonas\Desktop\test\one-click-hugo-cms [master ≡]> npm install
    npm WARN deprecated babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read babeljs.io/env to update!
    npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
    npm WARN deprecated postcss-cssnext@3.1.0: 'postcss-cssnext' has been deprecated in favor of 'postcss-preset-env'. Read more at https://moox.io/blog/deprecating-cssnext/
    npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
    npm WARN deprecated browserslist@2.11.3: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
    npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
    npm WARN deprecated natives@1.1.6: This module relies on Node.js's internals and will break at some point. Do not use it, and update to graceful-fs@4.x.
    npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
    npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
    npm WARN deprecated core-js@1.2.7: core-js@<2.6.5 is no longer maintained. Please, upgrade to core-js@3 or at least to actual version of core-js@2.    

    > uglifyjs-webpack-plugin@0.4.6 postinstall C:\Users\Jonas\Desktop\test\one-click-hugo-cms\node_modules\uglifyjs-webpack-plugin
    > node lib/post_install.js    

    npm notice created a lockfile as package-lock.json. You should commit this file.
    npm WARN css-loader@2.1.1 requires a peer of webpack@^4.0.0 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-app@2.9.1 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-backend-bitbucket@2.4.0 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-backend-test@2.2.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-backend-github@2.4.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-editor-component-image@2.4.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-core@2.11.0 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-backend-git-gateway@2.4.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-backend-gitlab@2.3.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-date@2.3.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-datetime@2.2.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-file@2.4.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-image@2.3.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-map@1.3.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-boolean@2.2.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-ui-default@2.6.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-markdown@2.4.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-list@2.3.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-number@2.3.3 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-object@2.2.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-relation@2.3.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-select@2.4.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-string@2.2.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN netlify-cms-widget-text@2.2.2 requires a peer of react@^16.8.4 but none is installed. You must install peer dependencies yourself.
    npm WARN react-dnd@7.4.5 requires a peer of react@>= 16.8 but none is installed. You must install peer dependencies yourself.
    npm WARN react-redux@4.4.10 requires a peer of redux@^2.0.0 || ^3.0.0 but none is installed. You must install peer dependencies yourself.
    npm WARN slate-edit-list@0.11.3 requires a peer of slate@^0.32.0 but none is installed. You must install peer dependencies yourself.
    npm WARN slate-edit-table@0.15.2 requires a peer of slate@^0.33.3 but none is installed. You must install peer dependencies yourself.
    npm WARN victor-hugo@1.0.0 No repository field.
    npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\fsevents):
    npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})    

    added 1682 packages from 941 contributors and audited 15065 packages in 169.565s
    found 8 vulnerabilities (1 low, 2 moderate, 5 high)
      run `npm audit fix` to fix them, or `npm audit` for details
    C:\Users\Jonas\Desktop\test\one-click-hugo-cms [master ≡ +1 ~0 -0 !]> npm install
    Batchvorgang abbrechen (J/N)? j
    C:\Users\Jonas\Desktop\test\one-click-hugo-cms [master ≡ +1 ~0 -0 !]> npm start    

    > victor-hugo@1.0.0 start C:\Users\Jonas\Desktop\test\one-click-hugo-cms
    > gulp server    

    [22:49:31] Failed to load external module @babel/register
    [22:49:31] Requiring external module babel-register
    fs.js:27
    const { Math, Object, Reflect } = primordials;
                                      ^    

    ReferenceError: primordials is not defined
        at fs.js:27:35
        at req_ (C:\Users\Jonas\Desktop\test\one-click-hugo-cms\node_modules\natives\index.js:143:24)
        at Object.req [as require] (C:\Users\Jonas\Desktop\test\one-click-hugo-cms\node_modules\natives\index.js:55:10)
        at Object.<anonymous> (C:\Users\Jonas\Desktop\test\one-click-hugo-cms\node_modules\vinyl-fs\node_modules\graceful-fs\fs.js:1:37)
        at Module._compile (internal/modules/cjs/loader.js:759:30)
        at Module._extensions..js (internal/modules/cjs/loader.js:770:10)
        at Object.require.extensions.<computed> [as .js] (C:\Users\Jonas\Desktop\test\one-click-hugo-cms\node_modules\babel-register\lib\node.js:152:7)
        at Module.load (internal/modules/cjs/loader.js:628:32)
        at Function.Module._load (internal/modules/cjs/loader.js:555:12)
        at Module.require (internal/modules/cjs/loader.js:666:19)
    npm ERR! code ELIFECYCLE
    npm ERR! errno 1
    npm ERR! victor-hugo@1.0.0 start: `gulp server`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the victor-hugo@1.0.0 start script.
    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.    

    npm ERR! A complete log of this run can be found in:
    npm ERR!     C:\Users\Jonas\AppData\Roaming\npm-cache\_logs\2019-05-15T20_49_32_119Z-debug.log

**Comment by jonas-sk** [comment]
I believe the bug relates to using gulp 3.x and Node 12 at the same time. I updated gulp and now it works (see PR #129)

**Comment by erquhart** [comment]
Just went through your PR's, thanks so much for digging in!

**Comment by erquhart** [comment]
@jonas-sk I just finished overhauling this repo by rebasing it onto [Victor Hugo](https://github.com/netlify-templates/victor-hugo), which it was originally based on. Victor Hugo has received much more support and maintenance from the community in the last two years, most notably being upgraded to Webpack 4 and removing Gulp entirely. Some improvements were made to templates as well, but most of them remain as they were.

I accidentally closed your PR's while bullk closing a bunch from the dependency management bot, but considering their age I decided to leave them closed and allow you to re-open any that you feel are still relevant against the latest updates.
